### PR TITLE
Setup Rector checks

### PR DESCRIPTION
**Pull request by alexndlm**

### CODE REVIEW

**Review by ERuban — APPROVED** [review]

File: composer.json [hunk 1]
```diff
@@ -1,45 +1,51 @@
...
-        }
-    },
-    "scripts": {
-        "check-all": "@parallel check-ecs",
-        "check-ecs": "php -d memory_limit=2048M vendor/bin/ecs check --clear-cache",
-        "fix-all": [
-            "@composer fix-ecs"
-        ],
-        "fix-ecs": "php -d memory_limit=2048M vendor/bin/ecs check --fix"
+  "name": "eonx-com/easy-quality",
```
Comment (roman-eonx) [inline composer.json:2]: Please revert the indents.

File: src/PHPStan/ThrowExceptionMessageRule.php [hunk 2]
```diff
@@ -19,19 +18,13 @@ final class ThrowExceptionMessageRule implements Rule
...
-     */
-    private $exceptionInterface;
-
     /**
      * @var string[]|null
      */
     private $validPrefixes;
 
-    public function __construct(?string $exceptionInterface = null, ?array $validPrefixes = null)
+    public function __construct(private readonly ?string $exceptionInterface = null, ?array $validPrefixes = null)
```
Comment (roman-eonx) [inline src/PHPStan/ThrowExceptionMessageRule.php:26]: Why not promote the second property as well? If I remember correctly, we allow default values in constructors.

File: src/Rector/PhpDocCommentRector.php [hunk 3]
```diff
@@ -204,7 +204,7 @@ private function checkIsMultilineNode(array $children): void
             }
 
             if ($value instanceof GenericTagValueNode) {
-                $containsEol = Strings::contains($value->value, \PHP_EOL);
+                $containsEol = \str_contains($value->value, \PHP_EOL);
                 $lastLetter = Strings::substring($value->value, -1, 1);
```
Comment (roman-eonx) [inline src/Rector/PhpDocCommentRector.php:208]: Can we replace this with a built-in function and get rid of `Strings` completely? Here and in other rector classes.

File: src/Sniffs/Classes/AvoidPublicPropertiesSniff.php [hunk 4]
```diff
@@ -45,14 +46,14 @@ protected function processMemberVar(File $phpcsFile, $stackPtr): void
      */
     protected function processVariable(File $phpcsFile, $stackPtr): void
     {
-        // No needed for sniff.
+        // No needed for sniff
```
Comment (roman-eonx) [inline src/Sniffs/Classes/AvoidPublicPropertiesSniff.php:49]: Please change to `Not needed`. Here and below.

File: src/Sniffs/Commenting/FunctionCommentSniff.php [hunk 5]
```diff
@@ -18,6 +10,13 @@
 use PHP_CodeSniffer\Util\Tokens;
 use SlevomatCodingStandard\Helpers\SuppressHelper;
 
+/**
```
Comment (roman-eonx) [inline src/Sniffs/Commenting/FunctionCommentSniff.php:13]: This comment is obsolete, let's get rid of it.

File: src/Sniffs/Commenting/FunctionCommentSniff.php [hunk 6]
```diff
@@ -192,16 +190,15 @@ protected function processParams(File $phpcsFile, $stackPtr, $commentStart): voi
                         // $error = 'Missing parameter comment';
```
Comment (roman-eonx) [inline src/Sniffs/Commenting/FunctionCommentSniff.php:190]: Let's get rid of the dead code.

File: src/Sniffs/Commenting/FunctionCommentSniff.php [hunk 7]
```diff
@@ -192,16 +190,15 @@ protected function processParams(File $phpcsFile, $stackPtr, $commentStart): voi
...
                         // $phpcsFile->addError($error, $tag, 'MissingParamComment');
                         $commentLines[] = ['comment' => ''];
-                    }//end if
+                    }// End if
```
Comment (roman-eonx) [inline src/Sniffs/Commenting/FunctionCommentSniff.php:193]: Let's remove all such useless comments.

**Review by roman-eonx — CHANGES_REQUESTED** [review]

File: src/PHPStan/ThrowExceptionMessageRule.php (hunk 2 above)
Comment (alexndlm) [reply src/PHPStan/ThrowExceptionMessageRule.php:26]: Super! Thanks! Done!

File: src/Rector/PhpDocCommentRector.php (hunk 3 above)
Comment (alexndlm) [reply src/Rector/PhpDocCommentRector.php:208]: With joy.

Done.

File: src/Sniffs/Classes/AvoidPublicPropertiesSniff.php (hunk 4 above)
Comment (alexndlm) [reply src/Sniffs/Classes/AvoidPublicPropertiesSniff.php:49]: Done.

File: src/Sniffs/Commenting/FunctionCommentSniff.php (hunk 5 above)
Comment (alexndlm) [reply src/Sniffs/Commenting/FunctionCommentSniff.php:13]: Done.

File: src/Sniffs/Commenting/FunctionCommentSniff.php (hunk 6 above)
Comment (alexndlm) [reply src/Sniffs/Commenting/FunctionCommentSniff.php:190]: Done.

File: src/Sniffs/Commenting/FunctionCommentSniff.php (hunk 7 above)
Comment (alexndlm) [reply src/Sniffs/Commenting/FunctionCommentSniff.php:193]: Done.

File: composer.json (hunk 1 above)
Comment (alexndlm) [reply composer.json:2]: Thanks. Fixed

File: src/Sniffs/Commenting/FunctionCommentSniff.php (hunk 6 above)
Comment (roman-eonx) [reply src/Sniffs/Commenting/FunctionCommentSniff.php]: All the commented code lines in this file, please.

File: src/Sniffs/Commenting/FunctionCommentSniff.php (hunk 7 above)
Comment (roman-eonx) [reply src/Sniffs/Commenting/FunctionCommentSniff.php]: At least one more left.

Comment (alexndlm) [reply src/Sniffs/Commenting/FunctionCommentSniff.php]: Sorry, I have rechecked and fixed all.

File: src/Sniffs/Commenting/FunctionCommentSniff.php (hunk 6 above)
Comment (alexndlm) [reply src/Sniffs/Commenting/FunctionCommentSniff.php]: Sorry, I have rechecked and fixed all.

**Review by roman-eonx — APPROVED** [review]

**Pull request merged**